ykman-gui/qml/YubiKey.qml

```diff
@@ -306,7 +342,36 @@ Python {
         do_call('yubikey.controller.fido_pin_retries', [], cb)
     }
 
+    function piv_change_pin(old_pin, new_pin, cb) {
+        do_call('yubikey.controller.piv_change_pin', [old_pin, new_pin],
+                _refreshPivBefore(cb))
+    }
+
+    function piv_change_puk(old_puk, new_puk, cb) {
+        do_call('yubikey.controller.piv_change_puk', [old_puk, new_puk],
+                _refreshPivBefore(function (resp) {
+                    if (!resp.success && resp.error === 'blocked') {
+                        pivPukBlocked = true
+                    }
+                    cb(resp)
+                }))
+    }
+
     function piv_reset(cb) {
```

Comment:
Why should reset refresh PIV before?

Reply:
The refresh is done before running the callback, after the Python call finishes.